File: Easy/206_reverse_linked_list.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* river = NULL;
        while(head != NULL) {
            ListNode* temp = new ListNode();
            temp->val = head->val;
            temp->next = river;
            river = temp;
            head = head->next;
        }
        return river;
    }
};

File: python/easy/206_reverse_linked_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev = None
        while head:
            temp = ListNode()
            temp.val = head.val
            temp.next = prev
            prev = temp
            head = head.next
        return prev
            
